examples/tv-casting-app/darwin/MatterTvCastingBridge/MatterTvCastingBridge/MCCastingPlayer.mm

```diff
@@ -41,53 +45,118 @@ + (NSInteger)kMinCommissioningWindowTimeoutSec
+        ChipLogError(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCallbacks() MCCastingApp NOT running"));
 
     dispatch_queue_t workQueue = [[MCCastingApp getSharedInstance] getWorkQueue];
     dispatch_sync(workQueue, ^{
-        matter::casting::core::IdentificationDeclarationOptions idOptions;
-
-        // TODO: In the following PRs. Replace EndpointFilter objC class with IdentificationDeclarationOptions objC class.
-        __block matter::casting::core::EndpointFilter cppDesiredEndpointFilter;
-        if (desiredEndpointFilter != nil) {
-            chip::Protocols::UserDirectedCommissioning::TargetAppInfo targetAppInfo;
-            targetAppInfo.vendorId = desiredEndpointFilter.vendorId;
-            targetAppInfo.productId = desiredEndpointFilter.productId;
-
-            CHIP_ERROR result = idOptions.addTargetAppInfo(targetAppInfo);
-            if (result != CHIP_NO_ERROR) {
-                ChipLogError(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCompletionBlock() failed to add targetAppInfo: %" CHIP_ERROR_FORMAT, result.Format());
-            }
+        matter::casting::core::IdentificationDeclarationOptions cppIdOptions;
+        if (identificationDeclarationOptions != nil) {
+            cppIdOptions = [identificationDeclarationOptions getCppIdentificationDeclarationOptions];
+        } else {
+            ChipLogProgress(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCallbacks()client did not set the optional MCIdentificationDeclarationOptions using default options");
         }
 
         void (^connectCallback)(CHIP_ERROR, matter::casting::core::CastingPlayer *) = ^(CHIP_ERROR err, matter::casting::core::CastingPlayer * castingPlayer) {
-            ChipLogProgress(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCompletionBlock() ConnectCallback()");
+            ChipLogProgress(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCallbacks() connectCallback() called");
+            dispatch_queue_t clientQueue = [[MCCastingApp getSharedInstance] getClientQueue];
+            dispatch_async(clientQueue, ^{
+                if (connectionCallbacks.connectionCompleteCallback) {
+                    connectionCallbacks.connectionCompleteCallback(err == CHIP_NO_ERROR ? nil : [MCErrorUtils NSErrorFromChipError:err]);
+                } else {
+                    ChipLogError(AppServer, "MCCastingPlayer.verifyOrEstablishConnectionWithCallbacks() connectCallback(), client failed to set the connectionCompleteCallback() callback");
+                }
+            });
+        };
+        void (^commissionerDeclarationCallback)(const chip::Transport::PeerAddress & source, const chip::Protocols::UserDirectedCommissioning::CommissionerDeclaration cppCommissionerDeclaration) = ^(const chip::Transport::PeerAddress & source, const chip::Protocols::UserDirectedCommissioning::CommissionerDeclaration cppCommissionerDeclaration) {
```

Comment:
Since we now have multiple callbacks, can you please add a couple of lines of inline comments here and for the other callback to say what it is that we are setting up?